src/fluree/db/storage/core.cljc

```diff
@@ -1,198 +1,99 @@
-   (let [leaf-id (random-leaf-id network ledger-id idx-type)]
-     (write-leaf conn network ledger-id idx-type leaf-id leaf)))
-
-  ([conn network ledger-id idx-type leaf-id {:keys [flakes] :as leaf}]
-   (go-try
-     (let [data {:flakes flakes}
-           ser  (serdeproto/-serialize-leaf (serde conn) data)
-           res  (<? (write conn leaf-id ser))]
-       (assoc leaf :id (or (:address res) leaf-id))))))
+  [{:keys [conn ledger] :as db} idx-type leaf]
+  (go-try
+    (let [ser (serdeproto/-serialize-leaf (serde conn) leaf)
+          res (<? (conn-proto/-index-file-write conn ledger idx-type ser))]
+      (log/warn "WRITE-LEAF COMPLETE WITH: " res)
+      res)))
 
 (defn write-branch-data
-  "Serializes final data for branch and writes it to provided key"
-  [conn key data]
+  "Serializes final data for branch and writes it to provided key.
+  Returns two-tuple of response output and raw bytes written."
+  [{:keys [conn ledger] :as db} idx-type data]
   (go-try
-    (let [ser (serdeproto/-serialize-branch (serde conn) data)]
-      (<? (write conn key ser)))))
-
-(defn random-branch-id
-  [network ledger-id idx]
-  (ledger-node-key network ledger-id idx (random-uuid) "b"))
+    (let [ser (serdeproto/-serialize-branch (serde conn) data)
+          res (<? (conn-proto/-index-file-write conn ledger idx-type ser))]
+      (log/warn "WRITE-BRANCH COMPLETE WITH: " res)
+      res)))
 
 (defn write-branch
   "Writes `branch` to storage under the provided `branch-id`, computing a new id
   if one isn't provided. Returns the branch map with the id used attached uner
   the `:id` key"
-  ([conn network ledger-id idx-type branch]
-   (let [branch-id (random-branch-id network ledger-id idx-type)]
-     (write-branch conn network ledger-id idx-type branch-id branch)))
-
-  ([conn network ledger-id idx-type branch-id {:keys [children] :as branch}]
-   (go-try
-     (let [child-vals  (->> children
-                            (map val)
-                            (mapv child-data))
-           first-flake (->> child-vals first :first)
-           rhs         (->> child-vals rseq first :rhs)
-           data        {:children child-vals}
-           res         (<? (write-branch-data conn branch-id data))]
-       (assoc branch :id (or (:address res) branch-id))))))
+  [db idx-type  {:keys [children] :as branch}]
+  (go-try
+    (let [child-vals  (->> children
+                           (map val)
+                           (mapv child-data))
+          data        {:children child-vals}
+          res         (<? (write-branch-data db idx-type data))]
+      (assoc branch :id (:address res)))))
 
 (defn write-garbage
   "Writes garbage record out for latest index."
   [db garbage]
   (go-try
-    (let [{:keys [conn network ledger-id block]} db
-          garbage-key (ledger-garbage-key network ledger-id block)
+    (let [{:keys [conn ledger ledger-id t]} db
+          t'          (- t) ;; use positive t integer
```

Comment:
Also not for this pr, but I think we should think about handling this conversion (and possibly others) in a more systematic way at the boundaries to make it less likely someone will for get to (de)apply it when dealing with `t` values.

Reply:
Perhaps it should be applied in the serialization / deserialization protocol.